src/toJS.js

```diff
@@ -2,8 +2,30 @@ import { Seq } from './Seq';
 import { isCollection } from './predicates/isCollection';
 import { isKeyed } from './predicates/isKeyed';
 import isDataStructure from './utils/isDataStructure';
+import isPlainObject from './utils/isPlainObj';
 
 export function toJS(value) {
+  const circularStack =
+    typeof WeakSet === 'undefined' ? new Set() : new WeakSet(); // WeakSet is not available in IE11.
+
+  return toJSWithCircularCheck(circularStack, value);
+}
+
+function checkCircular(circularStack, value) {
+  if (!isPlainObject(value)) {
+    return;
+  }
+
+  if (circularStack.has(value)) {
+    throw new TypeError('Cannot convert circular structure to JS');
```

Comment:
Instead of throwing here, how about returning `circularStack.get(value)` to recreate the cycle?